File: app/src/main/java/com/dbeginc/dbweather/ui/introduction/viewpager/fragment/IntroFragment.java
package com.dbeginc.dbweather.ui.introduction.viewpager.fragment;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by darel on 01.05.17.
 * Intro Fragment representing
 * introduction page
 */

public class IntroFragment extends Fragment {

    private int fragmentLayout;

    private static final String INTRO_LAYOUT = "intro_layout";

    public static IntroFragment newInstance(@LayoutRes final int backgroundRes) {

        final IntroFragment fragment = new IntroFragment();

        final Bundle data = new Bundle();

        data.putInt(INTRO_LAYOUT, backgroundRes);

        fragment.setArguments(data);

        return fragment;
    }

    @Override
    public void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            fragmentLayout = getArguments().getInt(INTRO_LAYOUT);
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull final LayoutInflater inflater, @Nullable final ViewGroup container, @Nullable final Bundle savedInstanceState) {
        return inflater.inflate(fragmentLayout, container, false);
    }
}

File: app/src/main/java/com/dbeginc/dbweather/launch/IntroFragment.kt
/*
 *  Copyright (C) 2017 Darel Bitsy
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License
 */

package com.dbeginc.dbweather.launch


import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.dbeginc.dbweather.R
import com.dbeginc.dbweather.databinding.FragmentIntroBinding
import com.dbeginc.dbweather.utils.utility.goToChooseLocationScreen
import com.dbeginc.dbweather.utils.utility.goToGpsLocationFinder


/**
 * A simple [Fragment] subclass.
 *
 */
class IntroFragment : Fragment() {
    private lateinit var binding: FragmentIntroBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        binding = DataBindingUtil.inflate(
                inflater,
                R.layout.fragment_intro,
                container,
                false
        )

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.chooseLocationBtn.setOnClickListener {
            activity?.let {
                goToChooseLocationScreen(container = it, layoutId = R.id.launchContent)
            }
        }

        binding.useGpsLocationBtn.setOnClickListener {
            activity?.let {
                goToGpsLocationFinder(container = it, layoutId = R.id.launchContent)
            }
        }
    }


}
